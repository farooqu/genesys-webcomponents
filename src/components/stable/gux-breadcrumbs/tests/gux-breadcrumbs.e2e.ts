import { newSparkE2EPage, a11yCheck } from '../../../../../tests/e2eTestUtils';

describe('gux-breadcrumbs', () => {
  [
    `<gux-breadcrumbs lang="en"></gux-breadcrumbs>`,
    `
      <gux-breadcrumbs lang="en">
        <gux-breadcrumb-item>Category</gux-breadcrumb-item>
        <gux-breadcrumb-item href="#">Parent</gux-breadcrumb-item>
        <gux-breadcrumb-item>Current</gux-breadcrumb-item>
      </gux-breadcrumbs>
    `,
    `
      <gux-breadcrumbs lang="en">
        <gux-breadcrumb-item>Category</gux-breadcrumb-item>
        <gux-breadcrumb-item href="#">Parent</gux-breadcrumb-item>
        <gux-breadcrumb-item href="#">Current</gux-breadcrumb-item>
      </gux-breadcrumbs>
    `,
    `
      <gux-breadcrumbs lang="en" accent="primary">
        <gux-breadcrumb-item>Category</gux-breadcrumb-item>
        <gux-breadcrumb-item href="#">Parent</gux-breadcrumb-item>
        <gux-breadcrumb-item>Current</gux-breadcrumb-item>
      </gux-breadcrumbs>
    `,
    `
      <gux-breadcrumbs lang="en" accent="secondary">
        <gux-breadcrumb-item>Category</gux-breadcrumb-item>
        <gux-breadcrumb-item href="#">Parent</gux-breadcrumb-item>
        <gux-breadcrumb-item>Current</gux-breadcrumb-item>
      </gux-breadcrumbs>
    `,
    `
      <gux-breadcrumbs lang="en" accent="invalid">
        <gux-breadcrumb-item>Category</gux-breadcrumb-item>
        <gux-breadcrumb-item href="#">Parent</gux-breadcrumb-item>
        <gux-breadcrumb-item>Current</gux-breadcrumb-item>
      </gux-breadcrumbs>
    `
  ].forEach((html, index) => {
    it(`should display component as expected (${index + 1})`, async () => {
      const page = await newSparkE2EPage({ html });

      const element = await page.find('gux-breadcrumbs');

      await a11yCheck(page);

      expect(element.outerHTML).toMatchSnapshot();
    });
  });
});
