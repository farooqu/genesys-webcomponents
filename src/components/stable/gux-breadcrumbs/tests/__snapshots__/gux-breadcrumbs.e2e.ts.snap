// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-breadcrumbs should display component as expected (1) 1`] = `"<gux-breadcrumbs lang=\\"en\\" class=\\"hydrated\\"><!----><nav aria-label=\\"Breadcrumbs\\" class=\\"gux-breadcrumbs-container\\"></nav></gux-breadcrumbs>"`;

exports[`gux-breadcrumbs should display component as expected (2) 1`] = `"<gux-breadcrumbs lang=\\"en\\" class=\\"hydrated\\"><!----><nav aria-label=\\"Breadcrumbs\\" class=\\"gux-breadcrumbs-container\\"> <gux-breadcrumb-item class=\\"hydrated\\"><!----><span class=\\"gux-breadcrumb-generation gux-primary\\"><span class=\\"gux-breadcrumb-content\\">Category</span><span class=\\"gux-breadcrumb-separator\\" aria-hidden=\\"true\\">/</span></span></gux-breadcrumb-item> <gux-breadcrumb-item href=\\"#\\" class=\\"hydrated\\"><!----><span class=\\"gux-breadcrumb-generation gux-primary\\"><a class=\\"gux-breadcrumb-content gux-link\\" href=\\"#\\">Parent</a><span class=\\"gux-breadcrumb-separator\\" aria-hidden=\\"true\\">/</span></span></gux-breadcrumb-item> <gux-breadcrumb-item class=\\"hydrated\\"><!----><span class=\\"gux-breadcrumb-generation gux-primary\\"><span class=\\"gux-breadcrumb-content\\">Current</span></span></gux-breadcrumb-item> </nav></gux-breadcrumbs>"`;

exports[`gux-breadcrumbs should display component as expected (3) 1`] = `"<gux-breadcrumbs lang=\\"en\\" class=\\"hydrated\\"><!----><nav aria-label=\\"Breadcrumbs\\" class=\\"gux-breadcrumbs-container\\"> <gux-breadcrumb-item class=\\"hydrated\\"><!----><span class=\\"gux-breadcrumb-generation gux-primary\\"><span class=\\"gux-breadcrumb-content\\">Category</span><span class=\\"gux-breadcrumb-separator\\" aria-hidden=\\"true\\">/</span></span></gux-breadcrumb-item> <gux-breadcrumb-item href=\\"#\\" class=\\"hydrated\\"><!----><span class=\\"gux-breadcrumb-generation gux-primary\\"><a class=\\"gux-breadcrumb-content gux-link\\" href=\\"#\\">Parent</a><span class=\\"gux-breadcrumb-separator\\" aria-hidden=\\"true\\">/</span></span></gux-breadcrumb-item> <gux-breadcrumb-item href=\\"#\\" class=\\"hydrated\\"><!----><span class=\\"gux-breadcrumb-generation gux-primary\\"><span class=\\"gux-breadcrumb-content\\">Current</span></span></gux-breadcrumb-item> </nav></gux-breadcrumbs>"`;

exports[`gux-breadcrumbs should display component as expected (4) 1`] = `"<gux-breadcrumbs lang=\\"en\\" accent=\\"primary\\" class=\\"hydrated\\"><!----><nav aria-label=\\"Breadcrumbs\\" class=\\"gux-breadcrumbs-container\\"> <gux-breadcrumb-item class=\\"hydrated\\"><!----><span class=\\"gux-breadcrumb-generation gux-primary\\"><span class=\\"gux-breadcrumb-content\\">Category</span><span class=\\"gux-breadcrumb-separator\\" aria-hidden=\\"true\\">/</span></span></gux-breadcrumb-item> <gux-breadcrumb-item href=\\"#\\" class=\\"hydrated\\"><!----><span class=\\"gux-breadcrumb-generation gux-primary\\"><a class=\\"gux-breadcrumb-content gux-link\\" href=\\"#\\">Parent</a><span class=\\"gux-breadcrumb-separator\\" aria-hidden=\\"true\\">/</span></span></gux-breadcrumb-item> <gux-breadcrumb-item class=\\"hydrated\\"><!----><span class=\\"gux-breadcrumb-generation gux-primary\\"><span class=\\"gux-breadcrumb-content\\">Current</span></span></gux-breadcrumb-item> </nav></gux-breadcrumbs>"`;

exports[`gux-breadcrumbs should display component as expected (5) 1`] = `"<gux-breadcrumbs lang=\\"en\\" accent=\\"secondary\\" class=\\"hydrated\\"><!----><nav aria-label=\\"Breadcrumbs\\" class=\\"gux-breadcrumbs-container\\"> <gux-breadcrumb-item class=\\"hydrated\\"><!----><span class=\\"gux-breadcrumb-generation gux-secondary\\"><span class=\\"gux-breadcrumb-content\\">Category</span><gux-icon class=\\"gux-breadcrumb-separator hydrated\\" icon-name=\\"chevron-sm-right\\"><div class=\\"gux-icon-container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-hidden=\\"true\\"><path fill-rule=\\"evenodd\\" clip-rule=\\"evenodd\\" d=\\"M11 8L7 4 6 5l3 3-3 3 1 1z\\"></path></svg></div></gux-icon></span></gux-breadcrumb-item> <gux-breadcrumb-item href=\\"#\\" class=\\"hydrated\\"><!----><span class=\\"gux-breadcrumb-generation gux-secondary\\"><a class=\\"gux-breadcrumb-content gux-link\\" href=\\"#\\">Parent</a><gux-icon class=\\"gux-breadcrumb-separator hydrated\\" icon-name=\\"chevron-sm-right\\"><div class=\\"gux-icon-container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-hidden=\\"true\\"><path fill-rule=\\"evenodd\\" clip-rule=\\"evenodd\\" d=\\"M11 8L7 4 6 5l3 3-3 3 1 1z\\"></path></svg></div></gux-icon></span></gux-breadcrumb-item> <gux-breadcrumb-item class=\\"hydrated\\"><!----><span class=\\"gux-breadcrumb-generation gux-secondary\\"><span class=\\"gux-breadcrumb-content\\">Current</span></span></gux-breadcrumb-item> </nav></gux-breadcrumbs>"`;

exports[`gux-breadcrumbs should display component as expected (6) 1`] = `"<gux-breadcrumbs lang=\\"en\\" accent=\\"invalid\\" class=\\"hydrated\\"><!----><nav aria-label=\\"Breadcrumbs\\" class=\\"gux-breadcrumbs-container\\"> <gux-breadcrumb-item class=\\"hydrated\\"><!----><span class=\\"gux-breadcrumb-generation gux-invalid\\"><span class=\\"gux-breadcrumb-content\\">Category</span><span class=\\"gux-breadcrumb-separator\\" aria-hidden=\\"true\\">/</span></span></gux-breadcrumb-item> <gux-breadcrumb-item href=\\"#\\" class=\\"hydrated\\"><!----><span class=\\"gux-breadcrumb-generation gux-invalid\\"><a class=\\"gux-breadcrumb-content gux-link\\" href=\\"#\\">Parent</a><span class=\\"gux-breadcrumb-separator\\" aria-hidden=\\"true\\">/</span></span></gux-breadcrumb-item> <gux-breadcrumb-item class=\\"hydrated\\"><!----><span class=\\"gux-breadcrumb-generation gux-invalid\\"><span class=\\"gux-breadcrumb-content\\">Current</span></span></gux-breadcrumb-item> </nav></gux-breadcrumbs>"`;
